### PR TITLE
CPU execution cap can be configured for the VirtualBox provider.

### PR DESCRIPTION
**Pull request by jdeathe**

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -83,6 +84,9 @@ Vagrant.configure("2") do |config|
         vb.gui = $vb_gui
         vb.memory = $vb_memory
         vb.cpus = $vb_cpus
+        if $vb_cpuexecutioncap
```
Comment (bcwaldon) [inline Vagrantfile]: What values of `$vb_cpuexecutioncap` is this conditional trying to catch?

Comment (jdeathe) [inline Vagrantfile]: Guess it could be removed, seeing as there is a default of 100 set.

Comment (bcwaldon) [inline Vagrantfile]: Ok, let's do that :)

**Comment by bcwaldon** [comment]
@jdeathe Please limit the subject of your commits message to ~60 chars, too

**Comment by jdeathe** [comment]
Okay sure :)

**Comment by jdeathe** [comment]
Edited commit message as requested.

File: Vagrantfile [hunk 2]
```diff
@@ -110,6 +111,7 @@ Vagrant.configure("2") do |config|
         vb.gui = vm_gui
         vb.memory = vm_memory
         vb.cpus = vm_cpus
+        vb.customize ["modifyvm", :id, "--cpuexecutioncap", "#{$vb_cpuexecutioncap}"]
```
Comment (bcwaldon) [inline Vagrantfile]: s/vb/vm/

**Comment by bcwaldon** [comment]
@jdeathe We need to support this for vmware, if possible. Have you looked into that at all?

**Comment by jdeathe** [comment]
@bcwaldon 
- The VirtualBox provider exposes the modifyvm command available parameters can be found in the documentation: https://www.virtualbox.org/manual/ch08.html#vboxmanage-modifyvm

- The VMware provider requires a license and documentation states the following:
> VMX is an undocumented format and there is no official reference for the available keys and values. This customization option is exposed for people who have knowledge of exactly what they want.

VMware has a CPU reservation setting (sched.cpu.min) however this does not directly compare to the execution cap setting found in VirtualBox and I'm not sure it can even be set if not a vmx.key-name type setting.

**Comment by bcwaldon** [comment]
@jdeathe ok, that's fair. Let's call this `vb_cpuexecutioncap` given that we're only applying it to the virtualbox provider.

**Pull request merged**